chainstate/test-suite/src/tests/pos_processing_tests.rs

```diff
@@ -0,0 +1,401 @@
+) -> (OutPoint, PoolId, VRFPrivateKey) {
+    let genesis_id = tf.genesis().get_id();
+    let (_, pub_key) = PrivateKey::new_from_rng(rng, KeyKind::RistrettoSchnorr);
+    let (vrf_sk, vrf_pk) = VRFPrivateKey::new(VRFKeyKind::Schnorrkel);
+    let tx = TransactionBuilder::new()
+        .add_input(
+            TxInput::new(OutPointSourceId::BlockReward(genesis_id.into()), 0),
+            empty_witness(rng),
+        )
+        .add_output(TxOutput::new(
+            OutputValue::Coin(Amount::from_atoms(1)),
+            OutputPurpose::StakePool(Box::new(StakePoolData::new(
+                anyonecanspend_address(),
+                None,
+                vrf_pk,
+                pub_key,
+                0,
+                Amount::ZERO,
+            ))),
+        ))
+        .build();
+    let tx_id = tx.transaction().get_id();
+    let pool_id = pos_accounting::make_pool_id(&OutPoint::new(
+        OutPointSourceId::BlockReward(genesis_id.into()),
+        0,
+    ));
+
+    tf.make_block_builder()
+        .add_transaction(tx)
+        .build_and_process()
+        .unwrap()
+        .unwrap();
+
+    (
+        OutPoint::new(OutPointSourceId::Transaction(tx_id), 0),
+        pool_id,
+        vrf_sk,
+    )
+}
+
+// Create a chain genesis <- block_1 <- block_1
+// PoS consensus activates on height 2.
+// block_1 has valid StakePool output. block_2 has PoS kernel input from block_1.
+// Check that the chain is valid.
+#[rstest]
+#[trace]
+#[case(Seed::from_entropy())]
+fn pos_basic(#[case] seed: Seed) {
+    let mut rng = make_seedable_rng(seed);
+
+    let upgrades = vec![
+        (
+            BlockHeight::new(0),
+            UpgradeVersion::ConsensusUpgrade(ConsensusUpgrade::IgnoreConsensus),
+        ),
+        (
+            BlockHeight::new(2),
+            UpgradeVersion::ConsensusUpgrade(ConsensusUpgrade::PoS),
+        ),
+    ];
+    let net_upgrades = NetUpgrades::initialize(upgrades).expect("valid net-upgrades");
+    let chain_config = ConfigBuilder::test_chain()
+        .net_upgrades(net_upgrades)
+        .epoch_length(NonZeroU64::new(2).unwrap())
+        .sealed_epoch_distance_from_tip(0)
+        .build();
+    let mut tf = TestFramework::builder(&mut rng).with_chain_config(chain_config).build();
+
+    let (stake_pool_outpoint, pool_id, vrf_sk) = create_chain_with_stake_pool(&mut rng, &mut tf);
+
+    let difficulty =
+        Uint256([0xFFFFFFFFFFFFFFFF, 0xFFFFFFFFFFFFFFFF, 0xFFFFFFFFFFFFFFFF, 0xFFFFFFFFFFFFFFFF]);
+    let prev_randomness = tf.chainstate.get_chain_config().initial_randomness();
+    let block_timestamp = BlockTimestamp::from_duration_since_epoch(tf.current_time());
+    let vrf_transcript = construct_transcript(1, &prev_randomness, block_timestamp);
+    let vrf_data = vrf_sk.produce_vrf_data(vrf_transcript.into());
+    let pos_data = PoSData::new(
+        vec![TxInput::from_outpoint(stake_pool_outpoint)],
+        vec![InputWitness::NoSignature(None)],
+        pool_id,
```

Comment:
You can in fact process the transaction and then check if the pool id that's used to store the accounting data matches the one you expect.